src/inmanta/env.py

```diff
@@ -185,30 +185,38 @@ def _gen_requirements_file(self, requirements_list: List[str]) -> str:
                     elif hasattr(item, "unsafe_name"):
                         name = item.unsafe_name
                     version = item.specs
+                    marker = item.marker
```

Comment:
`marker` should be reset to `None` before the try block, like `url` and `version`